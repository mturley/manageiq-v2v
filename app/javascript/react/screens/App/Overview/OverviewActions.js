import URI from 'urijs';
import API from '../../../../common/API';
import { V2V_NOTIFICATION_ADD } from '../common/NotificationList/NotificationConstants';
import { formatDateTime } from '../../../../components/dates/MomentDate';

import {
  ARCHIVE_TRANSFORMATION_PLAN,
  DELETE_TRANSFORMATION_PLAN,
  CREATE_V2V_TRANSFORMATION_PLAN_REQUEST,
  DELETE_INFRASTRUCTURE_MAPPING,
  FETCH_DATASTORES,
  FETCH_NETWORKS,
  FETCH_PROVIDERS,
  FETCH_PROVIDERS_URL,
  FETCH_V2V_ALL_ARCHIVED_PLAN_REQUESTS_WITH_TASKS,
  FETCH_V2V_SERVICE_TEMPLATE_ANSIBLE_PLAYBOOKS,
  FETCH_V2V_ALL_REQUESTS_WITH_TASKS,
  FETCH_V2V_ARCHIVED_TRANSFORMATION_PLANS,
  FETCH_V2V_TRANSFORMATION_MAPPINGS,
  FETCH_V2V_TRANSFORMATION_PLANS,
  HIDE_CONFIRM_MODAL,
  HIDE_DELETE_CONFIRMATION_MODAL,
  HIDE_MAPPING_WIZARD,
  SET_MAPPING_TO_DELETE,
  SHOW_CONFIRM_MODAL,
  SHOW_DELETE_CONFIRMATION_MODAL,
  SHOW_MAPPING_WIZARD,
  SHOW_PLAN_WIZARD,
  V2V_FETCH_CLUSTERS,
  V2V_RETRY_MIGRATION,
  V2V_SCHEDULE_MIGRATION,
  V2V_SET_MIGRATIONS_FILTER,
  V2V_TOGGLE_SCHEDULE_MIGRATION_MODAL,
  YES_TO_DELETE_AND_HIDE_DELETE_CONFIRMATION_MODAL,
  SHOW_PLAN_WIZARD_EDIT_MODE
} from './OverviewConstants';

export const showConfirmModalAction = modalOptions => ({
  type: SHOW_CONFIRM_MODAL,
  payload: modalOptions
});

export const hideConfirmModalAction = () => ({
  type: HIDE_CONFIRM_MODAL
});

export const showMappingWizardAction = () => dispatch => {
  dispatch({
    type: SHOW_MAPPING_WIZARD
  });
};

export const showPlanWizardAction = id => dispatch => {
  dispatch({
    type: SHOW_PLAN_WIZARD,
    payload: id
  });
};

export const showPlanWizardEditModeAction = id => dispatch => {
  dispatch({
    type: SHOW_PLAN_WIZARD_EDIT_MODE,
    editingPlanId: id
  });
};

export const fetchProvidersAction = () => dispatch => {
  dispatch({
    type: FETCH_PROVIDERS,
    payload: API.get(FETCH_PROVIDERS_URL)
  });
};

const _createTransformationPlanRequestActionCreator = url => dispatch =>
  dispatch({
    type: CREATE_V2V_TRANSFORMATION_PLAN_REQUEST,
    payload: {
      promise: API.post(url, { action: 'order' }),
      data: url
    }
  });

export const createTransformationPlanRequestAction = url => {
  const uri = new URI(url);
  return _createTransformationPlanRequestActionCreator(uri.toString());
};

const _getTransformationMappingsActionCreator = url => dispatch =>
  dispatch({
    type: FETCH_V2V_TRANSFORMATION_MAPPINGS,
    payload: API.get(url)
  });

export const fetchTransformationMappingsAction = url => {
  const uri = new URI(url);
  return _getTransformationMappingsActionCreator(uri.toString());
};

const fetchTasksForAllRequests = (allRequests, archived, dispatch) => {
  if (allRequests.length > 0) {
    dispatch({
      type: archived ? FETCH_V2V_ALL_ARCHIVED_PLAN_REQUESTS_WITH_TASKS : FETCH_V2V_ALL_REQUESTS_WITH_TASKS,
      payload: new Promise((resolve, reject) => {
        API.post('/api/requests?expand=resource&attributes=miq_request_tasks,service_template', {
          action: 'query',
          resources: allRequests
        })
          .then(responseRequestsWithTasks => {
            resolve(responseRequestsWithTasks);
          })
          .catch(e => reject(e));
      })
    });
  }
};

const collectAllRequests = plan => plan.miq_requests.map(request => Object.assign({}, { href: request.href }));

const _getTransformationPlansActionCreator = (url, archived) => dispatch =>
  dispatch({
    type: archived ? FETCH_V2V_ARCHIVED_TRANSFORMATION_PLANS : FETCH_V2V_TRANSFORMATION_PLANS,
    payload: new Promise((resolve, reject) => {
      API.get(url)
        .then(response => {
          resolve(response);
          const allPlansWithRequests = response.data.resources;

          const allRequests = [];
          const mergedRequests = [].concat(
            ...allRequests.concat(allPlansWithRequests.map(plan => collectAllRequests(plan)))
          );

          fetchTasksForAllRequests(mergedRequests, archived, dispatch);
        })
        .catch(e => reject(e));
    })
  });

export const fetchTransformationPlansAction = ({ url, archived }) => {
  const uri = new URI(url);
  return _getTransformationPlansActionCreator(uri.toString(), archived);
};

const _getServiceTemplateAnsiblePlaybooksActionCreator = url => dispatch =>
  dispatch({
    type: FETCH_V2V_SERVICE_TEMPLATE_ANSIBLE_PLAYBOOKS,
    payload: API.get(url)
  });

export const fetchServiceTemplateAnsiblePlaybooksAction = url => {
  const uri = new URI(url);
  return _getServiceTemplateAnsiblePlaybooksActionCreator(uri.toString());
};

export const continueToPlanAction = id => dispatch => {
  dispatch({
    type: HIDE_MAPPING_WIZARD
  });
  dispatch({
    type: SHOW_PLAN_WIZARD,
    payload: { id }
  });
};

const _getClustersActionCreator = url => dispatch =>
  dispatch({
    type: `${V2V_FETCH_CLUSTERS}`,
    payload: API.get(url)
  });

export const fetchClustersAction = url => {
  const uri = new URI(url);
  return _getClustersActionCreator(uri.toString());
};

const _getDatastoresActionCreator = url => dispatch =>
  dispatch({
    type: `${FETCH_DATASTORES}`,
    payload: API.get(url)
  });

export const fetchDatastoresAction = url => {
  const uri = new URI(url);
  return _getDatastoresActionCreator(uri.toString());
};

const _getNetworksActionCreator = url => dispatch =>
  dispatch({
    type: `${FETCH_NETWORKS}`,
    payload: API.get(url)
  });

export const fetchNetworksAction = url => {
  const uri = new URI(url);
  return _getNetworksActionCreator(uri.toString());
};

export const setMigrationsFilterAction = filter => ({
  type: V2V_SET_MIGRATIONS_FILTER,
  payload: filter
});

export const retryMigrationAction = planId => ({
  type: V2V_RETRY_MIGRATION,
  payload: planId
});

export const showDeleteConfirmationModalAction = () => ({
  type: SHOW_DELETE_CONFIRMATION_MODAL,
  payload: true
});

export const hideDeleteConfirmationModalAction = () => ({
  type: HIDE_DELETE_CONFIRMATION_MODAL,
  payload: false
});

export const setMappingToDeleteAction = mapping => dispatch => {
  dispatch({
    type: SET_MAPPING_TO_DELETE,
    payload: mapping
  });
};

export const yesToDeleteInfrastructureMappingAction = () => dispatch => {
  dispatch({
    type: YES_TO_DELETE_AND_HIDE_DELETE_CONFIRMATION_MODAL
  });
};

export const deleteInfrastructureMappingAction = mapping => dispatch =>
  dispatch({
    type: DELETE_INFRASTRUCTURE_MAPPING,
    payload: new Promise((resolve, reject) => {
      API.post(`/api/transformation_mappings/${mapping.id}`, {
        action: 'delete'
      })
        .then(response => {
          resolve(response);
        })
        .catch(e => reject(e));
    })
  });

const _archiveTransformationPlanActionCreator = url => dispatch =>
  dispatch({
    type: ARCHIVE_TRANSFORMATION_PLAN,
    payload: API.post(url, { action: 'archive' })
  });

export const archiveTransformationPlanAction = (url, id) => {
  const uri = new URI(`${url}/${id}`);
  return _archiveTransformationPlanActionCreator(uri.toString());
};

const _deleteTransformationPlanActionCreator = url => dispatch =>
  dispatch({
    type: DELETE_TRANSFORMATION_PLAN,
    payload: API.post(url, { action: 'delete' })
  });

export const deleteTransformationPlanAction = (url, id) => {
  const uri = new URI(`${url}/${id}`);
  return _deleteTransformationPlanActionCreator(uri.toString());
};

export const toggleScheduleMigrationModal = plan => ({
  type: V2V_TOGGLE_SCHEDULE_MIGRATION_MODAL,
  payload: plan
});

export const scheduleMigration = payload => dispatch =>
  dispatch({
    type: V2V_SCHEDULE_MIGRATION,
    payload: new Promise((resolve, reject) => {
      const {
        scheduleTime,
        plan: { id: planId }
      } = payload;
      const scheduleId = (payload.plan.schedules && payload.plan.schedules[0].id) || null;
      let url = `/api/service_templates/${planId}`;
      let body = {
        action: 'order',
        resource: {
          schedule_time: scheduleTime
        }
      };
      if (scheduleId) {
        url = `${url}/schedules/${scheduleId}`;
        body = scheduleTime
          ? {
              action: 'edit',
              resource: {
                run_at: { ...payload.plan.schedules[0].run_at, start_time: new Date(scheduleTime).toUTCString() }
              }
            }
          : { action: 'delete' };
      }
      return API.post(url, body)
        .then(response => {
          resolve(response);
          let msg = __('Migration successfully unscheduled');
          if (scheduleTime) {
            msg = sprintf(
              __('Migration successfully scheduled for %s'),
              formatDateTime(response.data.run_at.start_time)
            );
          }
          dispatch({
            type: V2V_NOTIFICATION_ADD,
            message: msg,
            notificationType: 'success',
            actionEnabled: false
          });
        })
        .catch(e => reject(e));
    })
  });
