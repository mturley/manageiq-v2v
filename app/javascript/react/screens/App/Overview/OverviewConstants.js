// reminder... we need to go back and prefix these: V2V_
export const SHOW_MAPPING_WIZARD = 'SHOW_MAPPING_WIZARD';
export const HIDE_MAPPING_WIZARD = 'HIDE_MAPPING_WIZARD';
export const MAPPING_WIZARD_EXITED = 'MAPPING_WIZARD_EXITED';
export const SHOW_PLAN_WIZARD = 'SHOW_PLAN_WIZARD';
export const SHOW_PLAN_WIZARD_EDIT_MODE = 'SHOW_PLAN_WIZARD_EDIT_MODE';
export const HIDE_PLAN_WIZARD = 'HIDE_PLAN_WIZARD';
export const PLAN_WIZARD_EXITED = 'PLAN_WIZARD_EXITED';
export const PLAN_WIZARD_NEXT = 'PLAN_WIZARD_NEXT';
export const PLAN_WIZARD_BACK = 'PLAN_WIZARD_BACK';
export const V2V_FETCH_CLUSTERS = 'FETCH_V2V_CLUSTERS';
export const CONTINUE_TO_PLAN = 'CONTINUE_TO_PLAN';
export const V2V_SET_MIGRATIONS_FILTER = 'V2V_SET_MIGRATIONS_FILTER';
export const V2V_RETRY_MIGRATION = 'V2V_RETRY_MIGRATION';
export const V2V_TOGGLE_SCHEDULE_MIGRATION_MODAL = 'V2V_TOGGLE_SCHEDULE_MIGRATION_MODAL';
export const V2V_SCHEDULE_MIGRATION = 'V2V_SCHEDULE_MIGRATION';
export const SHOW_CONFIRM_MODAL = 'SHOW_CONFIRM_MODAL';
export const HIDE_CONFIRM_MODAL = 'HIDE_CONFIRM_MODAL';
export const ARCHIVE_TRANSFORMATION_PLAN = 'ARCHIVE_TRANSFORMATION_PLAN';
export const DELETE_TRANSFORMATION_PLAN = 'DELETE_TRANSFORMATION_PLAN';

export const FETCH_PROVIDERS = 'FETCH_PROVIDERS';
export const FETCH_NETWORKS = 'FETCH_NETWORKS';
export const FETCH_DATASTORES = 'FETCH_DATASTORES';
export const FETCH_PROVIDERS_URL =
  '/api/providers?expand=resources&attributes=type&filter[]=or%20type=ManageIQ::Providers::Redhat::InfraManager&filter[]=type=ManageIQ::Providers::Vmware::InfraManager';

export const FETCH_V2V_TRANSFORMATION_MAPPINGS = 'FETCH_V2V_TRANSFORMATION_MAPPINGS';

export const FETCH_V2V_TRANSFORMATION_PLANS = 'FETCH_V2V_TRANSFORMATION_PLANS';
export const FETCH_V2V_SERVICE_TEMPLATE_ANSIBLE_PLAYBOOKS = 'FETCH_V2V_SERVICE_TEMPLATE_ANSIBLE_PLAYBOOKS';
export const FETCH_V2V_ARCHIVED_TRANSFORMATION_PLANS = 'FETCH_V2V_ARCHIVED_TRANSFORMATION_PLANS';

export const CREATE_V2V_TRANSFORMATION_PLAN_REQUEST = 'CREATE_V2V_TRANSFORMATION_PLAN_REQUEST';

export const FETCH_V2V_ALL_REQUESTS_WITH_TASKS = 'FETCH_V2V_ALL_REQUESTS_WITH_TASKS';
export const FETCH_V2V_ALL_ARCHIVED_PLAN_REQUESTS_WITH_TASKS = 'FETCH_V2V_ALL_ARCHIVED_PLAN_REQUESTS_WITH_TASKS';

export const SHOW_DELETE_CONFIRMATION_MODAL = 'SHOW_DELETE_CONFIRMATION_MODAL';
export const HIDE_DELETE_CONFIRMATION_MODAL = 'HIDE_DELETE_CONFIRMATION_MODAL';
export const SET_MAPPING_TO_DELETE = 'SET_MAPPING_TO_DELETE';
export const YES_TO_DELETE_AND_HIDE_DELETE_CONFIRMATION_MODAL = 'YES_TO_DELETE_AND_HIDE_DELETE_CONFIRMATION_MODAL';
export const DELETE_INFRASTRUCTURE_MAPPING = 'DELETE_INFRASTRUCTURE_MAPPING';

export const MIGRATIONS_FILTERS = {
  notStarted: __('Not Started Plans'),
  inProgress: __('In Progress Plans'),
  completed: __('Completed Plans'),
  archived: __('Archived Plans')
};

export const OSP_TENANT = 'CloudTenant';
export const OSP_VOLUME = 'CloudVolume';
export const OSP_NETWORK = 'CloudNetwork';
export const EMS_CLUSTER = 'EmsCluster';
