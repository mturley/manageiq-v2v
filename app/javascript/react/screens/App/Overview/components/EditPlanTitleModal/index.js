import { connect } from 'react-redux';
import EditPlanTitleModal from './EditPlanTitleModal';
import { findEditingPlan } from '../../screens/PlanWizard/PlanWizardSelectors';
import * as EditPlanTitleActions from './EditPlanTitleActions';

import reducer from './EditPlanTitleReducer';

export const reducers = { editPlanTitle: reducer };

const mapStateToProps = ({ editPlanTitle, overview, form: { editPlanTitleModal } }) => {
  const plans = [...overview.transformationPlans, ...overview.archivedTransformationPlans];
  const editingPlan = findEditingPlan(plans, overview.editingPlanId);
  return {
    ...editPlanTitle,
    editPlanTitleModal,
    initialValues: {
      name: editingPlan ? editingPlan.name : '',
      description: editingPlan ? editingPlan.description : ''
    },
    enableReinitialize: true,
    editingPlan
  };
};

export default connect(
  mapStateToProps,
  EditPlanTitleActions
)(EditPlanTitleModal);
