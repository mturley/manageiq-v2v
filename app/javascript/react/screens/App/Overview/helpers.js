import getMostRecentRequest from '../common/getMostRecentRequest';
import { TRANSFORMATION_MAPPING_ITEM_DESTINATION_TYPES } from '../Mappings/screens/MappingWizard/MappingWizardConstants';

export const planTransmutation = (plans = [], mappings = []) =>
  plans.map(plan => {
    const request = getMostRecentRequest(plan.miq_requests);
    return {
      ...plan,
      infraMappingName: plan.transformation_mapping && plan.transformation_mapping.name,
      status: request ? request.status : null,
      fulfilledOn: request ? request.fulfilled_on : null,
      configVmLength: plan.options.config_info.actions.length,
      scheduleTime: plan.schedules ? new Date(plan.schedules[0].run_at.start_time).getTime() : null
    };
  });

export const attachTargetProvider = (plan, providers, clusters, targetProviderType) => {
  if (!plan.transformation_mapping) {
    return plan;
  }

  const {
    transformation_mapping: { transformation_mapping_items }
  } = plan;
  const clusterMapping = transformation_mapping_items.find(
    item => item.destination_type === TRANSFORMATION_MAPPING_ITEM_DESTINATION_TYPES[targetProviderType].cluster
  );
  const targetCluster = clusters.find(cluster => cluster.id === clusterMapping.destination_id);
  if (targetCluster === undefined) {
    return plan;
  }
  const targetProvider = providers.find(provider => provider.id === targetCluster.ems_id);

  return { ...plan, targetProvider };
};

export const initializeDatepicker = (handleDatepickerChange, defaultDate) => {
  const datetimeSelector = $('#dateTimePicker');
  const minDate = new Date(Date.now() + 120000);

  datetimeSelector.datetimepicker({
    defaultDate: defaultDate > minDate ? defaultDate : minDate,
    useCurrent: !defaultDate,
    allowInputToggle: true,
    showTodayButton: true,
    minDate,
    toolbarPlacement: 'bottom',
    sideBySide: true,
    icons: {
      today: 'today-button-pf'
    }
  });

  datetimeSelector.on('dp.change', e => {
    handleDatepickerChange(e.date._d);
  });

  const picker = datetimeSelector.data('DateTimePicker');
  handleDatepickerChange(picker.date().toDate());
};
