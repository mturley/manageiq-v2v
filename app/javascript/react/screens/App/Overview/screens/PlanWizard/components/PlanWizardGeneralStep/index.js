import { connect } from 'react-redux';
import PlanWizardGeneralStep from './PlanWizardGeneralStep';
import { showAlertAction, hideAlertAction } from '../../PlanWizardActions';
import { findEditingPlan } from '../../PlanWizardSelectors';

const mapStateToProps = ({ overview }) => {
  const editingPlan = findEditingPlan(overview.transformationPlans, overview.editingPlanId);
  const prefilledMappingId =
    editingPlan &&
    editingPlan.options &&
    editingPlan.options.config_info &&
    editingPlan.options.config_info.transformation_mapping_id;
  return {
    transformationMappings: overview.transformationMappings,
    transformationPlans: overview.transformationPlans,
    archivedTransformationPlans: overview.archivedTransformationPlans,
    initialValues: {
      infrastructure_mapping: prefilledMappingId || overview.planWizardId,
      vm_choice_radio: 'vms_via_discovery',
      name: editingPlan ? editingPlan.name : '',
      description: editingPlan ? editingPlan.description : ''
    },
    enableReinitialize: true, // Tells redux-form to use new initialValues when they change
    editingPlan
  };
};

const actions = { showAlertAction, hideAlertAction };

export default connect(
  mapStateToProps,
  actions
)(PlanWizardGeneralStep);
