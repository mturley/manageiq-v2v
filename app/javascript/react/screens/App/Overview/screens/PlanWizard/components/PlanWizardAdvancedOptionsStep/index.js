import { connect } from 'react-redux';

import PlanWizardAdvancedOptionsStep from './PlanWizardAdvancedOptionsStep';
import * as PlanWizardAdvancedOptionsStepActions from './PlanWizardAdvancedOptionsStepActions';
import { getVMStepSelectedVms, getVmIdsWithProperty } from './helpers';
import reducer from './PlanWizardAdvancedOptionsStepReducer';
import { findEditingPlan } from '../../PlanWizardSelectors';

export const reducers = { planWizardAdvancedOptionsStep: reducer };

const mapStateToProps = (
  {
    planWizardAdvancedOptionsStep,
    planWizardVMStep,
    overview: { transformationPlans, editingPlanId },
    form: {
      planWizardGeneralStep: {
        values: { vm_choice_radio }
      },
      planWizardVMStep: {
        values: { selectedVms }
      },
      planWizardAdvancedOptionsStep: advancedOptionsStepForm
    }
  },
  ownProps
) => {
  const allVms =
    vm_choice_radio === 'vms_via_csv'
      ? [...planWizardVMStep.valid_vms, ...planWizardVMStep.invalid_vms, ...planWizardVMStep.conflict_vms]
      : [...planWizardVMStep.preselected_vms, ...planWizardVMStep.valid_vms];

  const editingPlan = findEditingPlan(transformationPlans, editingPlanId);
  const configInfo = editingPlan && editingPlan.options && editingPlan.options.config_info;

  return {
    ...planWizardAdvancedOptionsStep,
    ...ownProps.data,
    advancedOptionsStepForm,
    vmStepSelectedVms: getVMStepSelectedVms(allVms, selectedVms),
    initialValues: {
      playbookVms: {
        preMigration: editingPlan ? getVmIdsWithProperty(editingPlan, 'pre_service') : [],
        postMigration: editingPlan ? getVmIdsWithProperty(editingPlan, 'post_service') : []
      },
      preMigrationPlaybook: editingPlan ? configInfo.pre_service_id : '',
      postMigrationPlaybook: editingPlan ? configInfo.post_service_id : ''
    },
    enableReinitialize: true, // Tells redux-form to use new initialValues when they change
    keepDirtyOnReinitialize: true,
    editingPlan
  };
};

const mergeProps = (stateProps, dispatchProps, ownProps) => ({
  ...stateProps,
  ...ownProps.data,
  ...dispatchProps
});

export default connect(
  mapStateToProps,
  PlanWizardAdvancedOptionsStepActions,
  mergeProps
)(PlanWizardAdvancedOptionsStep);
