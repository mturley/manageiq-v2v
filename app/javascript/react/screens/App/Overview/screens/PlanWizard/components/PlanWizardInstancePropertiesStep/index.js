import { connect } from 'react-redux';

import PlanWizardInstancePropertiesStep from './PlanWizardInstancePropertiesStep';
import * as PlanWizardInstancePropertiesStepActions from './PlanWizardInstancePropertiesStepActions';
import reducer from './PlanWizardInstancePropertiesStepReducer';
import { getVMStepSelectedVms } from '../PlanWizardAdvancedOptionsStep/PlanWizardAdvancedOptionsStepSelectors';

export const reducers = { planWizardInstancePropertiesStep: reducer };

const mapStateToProps = (
  {
    overview: { transformationMappings },
    planWizardInstancePropertiesStep,
    planWizardVMStep,
    form: {
      planWizardGeneralStep: {
        values: { vm_choice_radio, infrastructure_mapping }
      },
      planWizardVMStep: {
        values: { selectedVms }
      },
      planWizardInstancePropertiesStep: instancePropertiesStepForm
    }
  },
  ownProps
) => {
  const allVms =
    vm_choice_radio === 'vms_via_csv'
      ? [...planWizardVMStep.valid_vms, ...planWizardVMStep.invalid_vms, ...planWizardVMStep.conflict_vms]
      : planWizardVMStep.valid_vms;
  const selectedMapping =
    transformationMappings &&
    infrastructure_mapping &&
    transformationMappings.find(mapping => mapping.id === infrastructure_mapping);
  return {
    ...planWizardInstancePropertiesStep,
    ...ownProps.data,
    instancePropertiesStepForm,
    vmStepSelectedVms: getVMStepSelectedVms(allVms, selectedVms),
    selectedMapping
  };
};

const mergeProps = (stateProps, dispatchProps, ownProps) => ({
  ...stateProps,
  ...ownProps.data,
  ...dispatchProps
});

export default connect(
  mapStateToProps,
  PlanWizardInstancePropertiesStepActions,
  mergeProps
)(PlanWizardInstancePropertiesStep);
