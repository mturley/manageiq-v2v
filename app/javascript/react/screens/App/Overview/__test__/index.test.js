import React from 'react';
import thunk from 'redux-thunk';
import promiseMiddleware from 'redux-promise-middleware';
import { createStore, applyMiddleware, combineReducers } from 'redux';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { initialState } from '../overview.fixtures';
import providersReducer, {
  initialState as providersInitialState
} from '../../../../../redux/common/providers/providersReducer';
import productFeaturesReducer, {
  initialState as productFeaturesInitialState
} from '../../../../../redux/common/productFeatures/productFeaturesReducer';
import Overview from '../Overview';
import OverviewContainer, { reducers } from '../index';

import { coreComponents } from '../../../../../components';
import componentRegistry from '../../../../../components/componentRegistry';

jest.mock('../../../../../components/componentRegistry');
componentRegistry.registerMultiple(coreComponents);

describe('Overview integration test', () => {
  const middlewares = [thunk, promiseMiddleware()];
  const generateStore = () =>
    createStore(
      combineReducers({ ...reducers, providers: providersReducer, productFeatures: productFeaturesReducer }),
      {
        overview: initialState,
        providers: providersInitialState,
        productFeatures: productFeaturesInitialState
      },
      applyMiddleware(...middlewares)
    );

  const mountComponent = store =>
    mount(
      <Provider store={store}>
        <OverviewContainer />
      </Provider>
    );

  it('should mount the Overview with mapStateToProps reduced', () => {
    const store = generateStore();
    const wrapper = mountComponent(store);

    // query the unconnected component to assert reduced props
    const component = wrapper.find(Overview);

    expect(component.props()).toMatchSnapshot();
  });
});
