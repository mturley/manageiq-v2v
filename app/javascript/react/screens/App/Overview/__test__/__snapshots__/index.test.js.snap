// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview integration test should mount the Overview with mapStateToProps reduced 1`] = `
Object {
  "acknowledgeDeniedPlanRequestAction": [Function],
  "activeTransformationPlans": Array [],
  "addNotificationAction": [Function],
  "archiveTransformationPlanAction": [Function],
  "archiveTransformationPlanUrl": "/api/service_templates",
  "archivedTransformationPlans": Array [],
  "cancelPlanRequestAction": [Function],
  "clusters": Array [],
  "createTransformationPlanRequestAction": [Function],
  "datastores": Array [],
  "deleteTransformationPlanAction": [Function],
  "deleteTransformationPlanUrl": "/api/service_templates",
  "editPlanRequestAction": [Function],
  "errorFetchingProviders": null,
  "fetchArchivedTransformationPlansUrl": "/api/dummyArchivedTransformationPlans",
  "fetchCloudTenantsAction": [Function],
  "fetchCloudTenantsUrl": "/api/cloud_tenants?expand=resources&filter[]=ext_management_system.type=ManageIQ::Providers::Openstack::CloudManager",
  "fetchClustersUrl": "/api/dummyClusters",
  "fetchDatastoresUrl": "/api/dummyDatastores",
  "fetchNetworksUrl": "/api/dummyNetworks",
  "fetchProductFeaturesAction": [Function],
  "fetchProductFeaturesActionUrl": "/api",
  "fetchProvidersAction": [Function],
  "fetchProvidersUrl": "/api/providers?expand=resources&attributes=authentications",
  "fetchServiceTemplateAnsiblePlaybooksAction": [Function],
  "fetchServiceTemplateAnsiblePlaybooksUrl": "/api/dummyServiceTemplatePlaybooks",
  "fetchTransformationMappingsAction": [Function],
  "fetchTransformationMappingsUrl": "/api/dummyMappings",
  "fetchTransformationPlansAction": [Function],
  "fetchTransformationPlansUrl": "/api/dummyTransformationPlans",
  "finishedTransformationPlans": Array [],
  "hasSufficientProviders": false,
  "hasTargetProvider": false,
  "hideConfirmModalAction": [Function],
  "hideEditPlanNameModalAction": [Function],
  "hideMappingWizard": false,
  "hidePlanWizard": false,
  "isFetchingClusters": false,
  "isFetchingDatastores": false,
  "isFetchingNetworks": false,
  "isFetchingProviders": true,
  "isFetchingTransformationMappings": false,
  "isRejectedClusters": false,
  "isRejectedDatastores": false,
  "isRejectedNetworks": false,
  "isRejectedProviders": false,
  "isRejectedTransformationMappings": false,
  "mappingWizardVisible": false,
  "networks": Array [],
  "notStartedTransformationPlans": Array [],
  "openMappingWizardOnTransitionAction": [Function],
  "planWizardVisible": false,
  "productFeatures": Array [],
  "providers": Array [],
  "redirectTo": [Function],
  "removeNotificationAction": [Function],
  "requestsProcessingCancellation": Array [],
  "retryMigrationAction": [Function],
  "scheduleCutover": [Function],
  "scheduleMigration": [Function],
  "setMigrationsFilterAction": [Function],
  "showConfirmModalAction": [Function],
  "showEditPlanNameModalAction": [Function],
  "showPlanWizardAction": [Function],
  "showPlanWizardEditModeAction": [Function],
  "toggleScheduleMigrationModal": [Function],
  "transformationMappings": Array [],
  "transformationPlans": Array [],
}
`;
